FordLabs/retroquest
# Java 17

# WIP DONOTMERGE

# This branch contains all the changes from the Angular deprecation branch. At the very least, this branch should not be merged into main until after the Angular deprecation branch is merged in. 

## Overview

Update RetroQuest to Java 17 and use new language features

### Notes
We already take advantage of [Gradle Toolchains](https://docs.gradle.org/current/userguide/toolchains.html), so this should have no impact to any users who would like to run RetroQuest themselves. This will not create a requirement that users must have a Java 17 JDK installed on their machine.

## Testing Instructions
- Run existing tests and ensure they still pass (there will be no functionality changes as a part of this pull request)
